docs/content/config/advanced/podman.md

```diff
@@ -96,6 +96,107 @@ docker compose up -d mailserver
+
+[podman-docs::cli::generate-systemd]: https://docs.podman.io/en/latest/markdown/podman-generate-systemd.1.html
+[podman-docs::quadlet::example-configs]: https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#examples
+[podman-docs::config::quadlet-generation]: https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#description
+[podman::quadlet::introduction]: https://mo8it.com/blog/quadlet/
+[podman::quadlet::generated-output-example]: https://blog.while-true-do.io/podman-quadlets/#writing-quadlets
+
+!!! tip "Rootless compatibility"
+
+    Quadlets can [support rootless with a few differences][podman::rootless-differences]:
+
+    - `Network=pasta` configures [`pasta`][network-driver::pasta] as a rootless compatible network driver (_a popular alternative to `slirp4netns`. `pasta` is the default for rootless since Podman v5_).
+    - `Restart=always` will auto-start your Quadlet at login, rootless support requires to enable [lingering][systemd-docs::loginctl::linger] for your user:
+
+        ```bash
+        loginctl enable-linger user
+        ```
+    - [Config locations between rootful vs rootless][podman-docs::quadlet::config-search-path].
+
+[podman::rootless-differences]: https://matduggan.com/replace-compose-with-quadlet/#rootless
+[network-driver::pasta]: https://passt.top/passt/about/#pasta
+[systemd-docs::loginctl::linger]: https://www.freedesktop.org/software/systemd/man/latest/loginctl.html#enable-linger%20USER%E2%80%A6
+[podman-docs::quadlet::config-search-path]: https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#podman-rootful-unit-search-path
+
+
+#### Example Quadlet file
+
+We have to use the .container extension for the quadlet generator to pick up the service.
+Because docker-mailserver uses multiple users inside the container, we will either have to use our own user as root, resulting in our e-mails being owned by a subuid. Alternatively, using UIDMap we can map our rootless user to UID 5000 in the container who owns our e-mails. Using UIDMap also maps root user 0 inside the container to an available sub-uid of our rootless user. Otherwise the container will not have permission to configure itself.
+
+The example uses `Network=pasta` to use the pasta network driver, which will replace `slirp4netns`.
+
+`dockermailservice.container`
+
+```
+[Service]
+Restart=always
+
+[Install]
+WantedBy=default.target
+
+[Unit]
+Wants=network-online.target 
+After=network-online.target
+
+[Container]
+ContainerName=dms
+Image=docker.io/mailserver/docker-mailserver:latest 
+# DMS uses uid 5000 for mailstate, but creates other folders for different users, which will be mapped to different sub-uids
+UIDMap=5000:0:1
+UIDMap=0:1:5000
+UIDMap=5001:5001:60536
```

Comment:
Awaiting your feedback before I make a suggestion here. I'd still like to document `UIDMap` since it did take a while to grok and make sense of and I've seen others confused by it when I was looking into how the mapping works (_volumes also map in reverse which was a bit unexpected, so at one point my host user resolved to intermediate ID `0` which then mapped to container ID 1 (`daemon`), or in the case of config above 5000 (`docker`)_).

Also, I'm not sure of the exact cause but I noticed that `UIDMap=+0:0` and `UIDMap=+0:1` were quick to start, but when the intermediate ID was higher than that there was a 25-30 second delay to container startup, presumably some sort of `chown` (this is on SSD storage too)? Your suggested config here with mapping `5000` to `0` also triggers that IIRC (_not this specific mapping, but something about the other lines that come after_).

Seems to be related to range mapping, as you can do a single mapping for `0` intermediate ID just fine, except for a `supervisord` failure with it's socket (_but this seems to be resolved when `65534` for `nobody` / `nogroup` ID is mapped too_).

Reply:
**TL;DR:** You can ignore the bulk of this, it's mostly for me or anyone else that needs the additional context if the topic comes up again in future.

`UIDMap=+0:@%U` seems like the better choice to default to for rootless podman needs.

---

## `UIDMap` notes | Mapping behaviour + chown delay and storage accumulation

It only seems like the delay was a one off hit per mapping. Each one was copying the image (700MB+), presumably entirely remapped to `~/.local/share/containers/storage/overlay`.

Here is the initial pull without any `UIDMap` involved:

![image](https://github.com/user-attachments/assets/a83d0c61-b22c-48d6-a7b1-47dd21c13acc)

```
UIDMap=5000:0:1
UIDMap=0:1:5000
UIDMap=5001:5001:60536
```

Is equivalent to `UIDMap=+5000:@%U` in mapping and thus once one is used, switching config to the other won't have startup delay as the overlay copy is already handled.

```console
$ podman exec -it dms cat /proc/self/uid_map

         0          1       5000
      5000          0          1
      5001       5001      60536
```

---

### ClamAV (UID 200) example | When mapping triggers a larger chown copy

The overhead from that mapping results in a overlay squashed layer weighing `762.56 MiB` as opposed to `538.51 MiB` for `UIDMap=+0:@%U`, with the main difference in weight being the ClamAV layer (223 MiB) not being included in the squashed overlay diff.

```console
$ podman exec -it dms bash -c 'ls -l /var/lib/ | grep clamav && ls -l /var/lib/clamav'

drwxr-xr-x. 1 clamav       clamav       4096 Jun  3 01:07 clamav
total 228992
-rw-r--r--. 1 clamav clamav    289733 Jun  3 01:07 bytecode.cvd
-rw-r--r--. 1 clamav clamav  63700823 Jun  3 01:06 daily.cvd
-rw-r--r--. 1 clamav clamav 170479789 Jun  3 01:06 main.cvd
```

`clamav` user and group have UID/GID of `200` within the container.

```console
# An extra `chown` copy happens with `UIDMap=+0:201`:
$ podman exec -it dms cat /proc/self/uid_map

         0        201          1
         1          0          1
         2          1        200
       202        202      65335

# As opposed to `UIDMap=+0:200`:
$ podman exec -it dms cat /proc/self/uid_map
         0        200          1
         1          0          1
         2          1        199
       201        201      65336

# Or  `UIDMap=+0:199`:
$ podman exec -it dms cat /proc/self/uid_map
         0        199          1
         1          0          1
         2          1        198
       200        200      65337
```

- The separate `1 0 1` mapping I think is added when you map with `+` prefix and seems a bit redundant in this case. I think it's implemented to handle the container root ID separately to avoid this error `container ID 0 cannot be mapped to a host ID`, while also handling any before/after mapping.
- `UIDMap=+0:201` => The `2 1 200` line covers UID 200 (clamav) thus it is mapped with an offset requiring a `chown`.
- `UIDMap=+0:200` => AFAIK there is no mapping for container UID 200 to any host ID, so it is unaffected.
-  `UIDMap=+0:199` => There is a mapping now, but it's direct `200` => `200`, it will use the existing overlay layer (_created during original pull_) which has that ownership with `chown` already applied for that offset:

   ```console
   $ ls -l /home/linuxuser/.local/share/containers/storage/overlay/ccab93e94733e94b645eb852f9bfa7471955cb1d184ef6224fbe51980d30e767/diff/var/lib/clamav
   total 228980
   -rw-r--r--. 1 524487 524487    289733 Jun  3 01:07 bytecode.cvd
   -rw-r--r--. 1 524487 524487  63700823 Jun  3 01:06 daily.cvd
   -rw-r--r--. 1 524487 524487 170479789 Jun  3 01:06 main.cvd
   ```
   
   `524487` is the base `524288` + (200 -1). UID 0 technically maps to 1000 (`linuxuser` in this case), then from UID offset of 1 in `/etc/subuid` on the host we increment from `524288`, hence container UID `200` maps to host UID `524487`:
   
   ```console
   $ cat /etc/subuid
   
   linuxuser:524288:65536
   ```

---

### Default mapping vs `UIDMap=+0:@1000`

For reference this is the default without any `UIDMap` setting:

```console
$ podman exec -it dms cat /proc/self/uid_map

         0       1000          1
         1     524288      65536
```

That is valid for rootful where the 2nd column would align with the host UID, but in rootless the 2nd column is a 
"intermediate ID" that is mapped as an offset for the user host ID entry in `/etc/subuid`.

Hence the `@1000` syntax for `UIDMap` which would be the equivalent for rootless (mapping to the `0` offset), so `UIDMap=+0:@1000` would get us the desired rootless mapping:

```console
$ podman exec -it dms cat /proc/self/uid_map

         0          0          1
         1          1      65536
```

### `chown` storage copies cleanup

The extra copies of the image at `~/.local/share/containers/storage/overlay` are tied to the image, so when all references are removed via `podman image rm` or `podman image prune -a` / `podman system prune -a` (_assuming no container is running with the image_).

I'm not aware of how you can clear that more selectively than that (_`podman run --rm --uidmap ...` is the same with accumulation_).